### ISSUE TITLE
Include response code in error objects

### ISSUE THREAD

**Issue by Nmargolis** [mapbox/HecateJS#15]
## Context
Currently, all of the methods in /lib return [new errors](https://github.com/mapbox/HecateJS/blob/master/lib/bbox.js#L93) with the response body as the error message. As an end user, it would be helpful to have access to the status code, instead of having to parse the response body to try to get to it. For example, we would like to catch `502` errors so we can try those requests later.

For a 502 error, the response body looks like this:
```<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
</body>
</html>
```

## Proposed change
Currently, errors are returned like this:
```
request({ ... }, (err, res) => {
    if (err) return cb(err);
    if (res.statusCode !== 200) return cb(new Error(res.body));
    ...
});
```

It would be great if they were returned like this:
```
request({ ... }, (err, res) => {
    if (err) return cb(err);
    if (res.statusCode !== 200) {
        const error = new Error(res.body);
        error.status = res.statusCode;
        return cb(error);
    }
    ...
});
```

cc @ingalls

**Comment by ingalls** [comment]
@Nmargolis are you working on this in the branch `error-status-codes`?

**Comment by Nmargolis** [comment]
@ingalls yes it's in the WIP PR #16. Would love your thoughts on the approach!
